### PR TITLE
Wobble in LST db

### PR DESCRIPTION
**Pull request by Elisa-Visentin**

### CODE REVIEW

**Comment by Elisa-Visentin** [comment]
Seems to work fine (I tested it on approx 1 y of data). Output is a new column in LST database 
image

**Comment by Elisa-Visentin** [comment]
We need only the offset (due to the IRFS, which are 0.4 point-like)

I retrieve the wobble from MAGIC file names (looking for a '-W after the source name), so, hopefully, it should be 0 (or not written in the name of the file with a -W pattern: I found some cases like that, but I have to investigate them) for ON obs.

**Comment by aleberti** [comment]
> We need only the offset (due to the IRFS, which are 0.4 point-like)

ok

> I retrieve the wobble from MAGIC file names (looking for a '-W after the source name), so, hopefully, it should be 0

better to check that it is actually 0 and not else. 

> (or not written in the name of the file with a -W pattern: I found some cases like that, but I have to investigate them) for ON obs.

it would be interesting to have cases where the patter for wobble is different from -WY.YY+ZZZ ... if you find some like that, please let me know.

**Comment by Elisa-Visentin** [comment]
Of course, in case some wobble obs. miss this pattern, they won't be 'listed' for the wobble evaluation...

**Comment by aleberti** [comment]
> I was just checking SGR: the output of the script is '[]', so quite easily identifiable. If DAQ SW never uses the pattern '{source}-W' for ON obs. the script never associate a float/list of floats to them

ok, then I agree it is straightforward to get ON observations. Correct, the DAQ SW will use the pattern `{source}` when ON observations are performed. Of course there is always a way to have ON observations and have the pattern `{source}-WY.YY+ZZZ` in the filename, but that requires shifters to input 0 for wobble offset and all the wobble angles. I think this never happened since I am in MAGIC (I could be wrong, of course).

**Comment by Elisa-Visentin** [comment]
I removed one of the input DBs because they have a slightly different format and 2023 processing failed

in case they set weobble=0, the file will have a pattern {source}-W0.00+* and the output will be '[0.00]' (in the hp that the wobble offset did not change during an LST run, which seems fine up to now)

**Comment by Elisa-Visentin** [comment]
> it would be interesting to have cases where the patter for wobble is different from -WY.YY+ZZZ ... if you find some like that, please let me know.
run 05109639, '--W....' 😕

**Comment by aleberti** [comment]
that would be a problem if for some reason the source name ends exactly with `-W`. Then, to guard yourself against this (unlikely but possible) case, you cannot simply search for the pattern `-W`. The most solid approach would be to search for the pattern `-WY.YY+ZZZ` via a regex. If that pattern is found, then you have the wobble information. If not, it is ON.

**Comment by Elisa-Visentin** [comment]
actually, also if source name ends with '-' I don't consider that '-' while looking for -W
While if I search '-WY.YY+Z.ZZ' over the while file name I could find the wrong match/wobble if the source has the '-WY.YY+Z.ZZ' INSIDE its name (this would be quite a strange name for a source, but not impossible at least in theory)

**Comment by aleberti** [comment]
I checked into the MAGIC database and I do not find sources with such a pattern. I found sources with `-W` in the name, but last time they were observed was in August 2019, so before the first date of joint observation in our table. In any case, since you get the source names from "your" database, then even if the whole wobble pattern were present in the source name, you would get the wobble information correctly.

**Review by aleberti — APPROVED** [review]

**Review by jsitarek — COMMENTED** [review]
just a tiny suggestion to simplify a bit the code, otherwise it is fine with me and we can merge

File: magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/wobble_db.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            .replace(" ", "")
+            .split(",")
+        )
+        lst_run = (df["LST1_run"].to_list())[i]
+        wobble = []
+        source = (df["source"].to_list())[i]
+        for j in range(len(magic_runs)):
+            print("MAGIC run:", magic_runs[j])
+            runs = glob.glob(
+                f"/fefs/onsite/common/MAGIC/data/M1/event/Calibrated/{date_magic[i]}/*{magic_runs[j]}*{source}*.root"
```
[the remark this answers is not in the document]
Comment (Elisa-Visentin) [reply magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/wobble_db.py:99]: mhh I have to check it in ipython just to be sure, but probably yes

**Pull request merged**